source/CustomSale.h

```diff
@@ -0,0 +1,96 @@
+/* CustomSale.h
+Copyright (c) 2021 by Hurleveur
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef CustomSale_H_
+#define CustomSale_H_
+
+#include "ConditionSet.h"
+#include "Outfit.h"
+#include "LocationFilter.h"
+#include "Sale.h"
+#include "Set.h"
+
+#include <map>
+#include <set>
+
+class ConditionsStore;
+class DataNode;
+class Planet;
+
+
+
+// Class used to stock Outfits and their local changes, being prices and sell types,
+// linked by outfit or by group of outfits (aka outfitters).
+class CustomSale {
+public:
+	// Sell types: none is meant to be default, signifying this is empty,
+	// visible is the default one in the game,
+	// import means you can only sell it but not buy it (and it is shown still)
```

Comment:
more of a request that can also be done later on: could we also do a case where you can not sell it?

Reply:
Meaning you can only buy? Sure, why not but in another PR.
I'll also add this for ships and allowing to sell a given amount onl per day etc